src/apify/_actor.py

```diff
@@ -974,6 +977,22 @@ async def create_proxy_configuration(
 
         return proxy_configuration
 
+    @staticmethod
```

Comment:
_ActorType class seems very large. Maybe we should compose it from smaller components instead of having big class with so many methods?

Reply:
You're right, feel free to open an issue for that. We better do a similar stuff in the JS SDK as well too.

Reply:
I wouldn't be so sure about this. `Actor` class is the main touchpoint, it consolidates all the public API, and moving the implementation somewhere else would only complicate development for us probably.

Not sure if python has traits like PHP does, JS doesn't have a thing like that, and even if it would have them, I don't think it would help much.

Reply:
Python has multiple inheritance, so we could have something like traits so that there is some internal division of the public methods.

Thanks for saying that we shouldn't break the public API @B4nan, I assumed the comment was about internal changes.

Reply:
I am still not convinced it would help. I did something like this in one project years ago, just to reduce the size of a class, I introduced unnecessary abstraction, and it is a pain in the ass to maintain it.

Let's write down some concrete ideas and decide on that.

Reply:
And please let's do so in a different issue and not derail this PR :slightly_smiling_face: